tests/monitoring/metrics.go

```diff
@@ -0,0 +1,154 @@
+			"kubevirt_vmi_migrations_in_scheduling_phase":                        true,
+			"kubevirt_vmi_migrations_in_running_phase":                           true,
+			"kubevirt_vmi_migration_succeeded":                                   true,
+			"kubevirt_vmi_migration_failed":                                      true,
+		}
+
+		It("should contain virt-operator metrics", func() {
+			err = virtoperator.SetupMetrics()
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, metric := range virtoperator.ListMetrics() {
+				Expect(metrics.Data.Result).To(ContainElement(gomegaContainsMetricMatcher(metric, nil)))
+			}
+		})
+
+		It("should contain virt-api metrics", func() {
+			err = virtapi.SetupMetrics()
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, metric := range virtapi.ListMetrics() {
+				if excludedMetrics[metric.GetOpts().Name] {
+					continue
+				}
+
+				Expect(metrics.Data.Result).To(ContainElement(gomegaContainsMetricMatcher(metric, nil)))
+			}
+		})
+
+		It("should contain virt-controller metrics", func() {
+			err = virtcontroller.SetupMetrics(nil, nil, nil, nil, nil, nil, nil, nil)
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, metric := range virtcontroller.ListMetrics() {
+				if excludedMetrics[metric.GetOpts().Name] {
+					continue
+				}
+
+				Expect(metrics.Data.Result).To(ContainElement(gomegaContainsMetricMatcher(metric, nil)))
+			}
+		})
+
+		Context("perfscale histogram metrics", func() {
+			It("should have kubevirt_vmi_phase_transition_time_seconds buckets correctly configured", func() {
+				buckets := virtcontroller.PhaseTransitionTimeBuckets()
+
+				for _, bucket := range buckets {
+					labels := map[string]string{"le": strconv.FormatFloat(bucket, 'f', -1, 64)}
+
+					metric := operatormetrics.NewHistogram(
+						operatormetrics.MetricOpts{Name: "kubevirt_vmi_phase_transition_time_from_deletion_seconds"},
+						prometheus.HistogramOpts{},
+					)
+
+					Expect(metrics.Data.Result).To(ContainElement(gomegaContainsMetricMatcher(metric, labels)))
+				}
+			})
+		})
+	})
+})
+
+func gomegaContainsMetricMatcher(metric operatormetrics.Metric, labels map[string]string) types.GomegaMatcher {
+	return &libmonitoring.MetricMatcher{Metric: metric, Labels: labels}
+}
+
+func setupVM(virtClient kubecli.KubevirtClient) {
+	vm := createRunningVM(virtClient)
+	libmonitoring.WaitForMetricValue(virtClient, "kubevirt_number_of_vms", 1)
+
+	By("Deleting the VirtualMachine")
+	err := virtClient.VirtualMachine(vm.Namespace).Delete(context.Background(), vm.Name, &metav1.DeleteOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	libmonitoring.WaitForMetricValue(virtClient, "kubevirt_number_of_vms", -1)
+}
+
+func createRunningVM(virtClient kubecli.KubevirtClient) *v1.VirtualMachine {
+	vmi := libvmi.NewCirros(
+		libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+		libvmi.WithNetwork(v1.DefaultPodNetwork()),
+	)
+	vmi.Namespace = testsuite.GetTestNamespace(nil)
```

Comment:
You can call `WithNamespace` option in `NewCirros`

Reply:
updated